components/org.wso2.carbon.identity.oauth.ciba/src/test/java/org/wso2/carbon/identity/oauth/ciba/dao/CibaMgtDAOImplTest.java

```diff
@@ -0,0 +1,416 @@
+        }
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            mockStatic(IdentityDatabaseUtil.class);
+            when(IdentityDatabaseUtil.getDBConnection(false)).thenReturn(connection1);
+            assertEquals(cibaMgtDAO.getCibaAuthCodeWithAuthReqID(AUTH_REQ_ID).getAuthenticationStatus(),
+                    AuthReqStatus.DENIED);
+        }
+    }
+
+    @Test
+    public void testPersistAuthenticatedUser() throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(mockedServerConfig);
+        mockStatic(OAuth2Util.class);
+        when(OAuth2Util.getTenantDomain(anyInt())).thenReturn("super.wso2");
+
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            mockStatic(IdentityDatabaseUtil.class);
+            when(IdentityDatabaseUtil.getDBConnection(true)).thenReturn(connection1);
+            cibaMgtDAO.persistAuthenticatedUser(AUTH_CODE_KEY, authenticatedUser, -1234);
+        }
+
+        try (Connection connection2 = getConnection(DB_NAME)) {
+            mockStatic(IdentityDatabaseUtil.class);
+            when(IdentityDatabaseUtil.getDBConnection(false)).thenReturn(connection2);
+            assertEquals(cibaMgtDAO.getAuthenticatedUser(AUTH_CODE_KEY).getUserName(), "randomUser");
+        }
+    }
+
+    @Test
+    public void testPersistAuthenticationSuccessStaus() throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            mockStatic(IdentityDatabaseUtil.class);
+            when(IdentityDatabaseUtil.getDBConnection(true)).thenReturn(connection1);
+            cibaMgtDAO.updateStatus(AUTH_CODE_KEY, AuthReqStatus.AUTHENTICATED);
+        }
+
+        try (Connection connection2 = getConnection(DB_NAME)) {
+            mockStatic(IdentityDatabaseUtil.class);
+            when(IdentityDatabaseUtil.getDBConnection(false)).thenReturn(connection2);
+            assertEquals(cibaMgtDAO.getCibaAuthCodeWithAuthReqID(AUTH_REQ_ID).getAuthenticationStatus(),
+                    AuthReqStatus.AUTHENTICATED);
+        }
+    }
+
+    @Test
+    public void testIsAuthReqIdExist() throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            mockStatic(IdentityDatabaseUtil.class);
+            when(IdentityDatabaseUtil.getDBConnection(false)).thenReturn(connection1);
+            assertTrue(cibaMgtDAO.isAuthReqIdExist(AUTH_REQ_ID));
+        }
+    }
+
+    @Test
+    public void testGetCibaAuthCodeKey() throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        try (Connection connection1 = getConnection(DB_NAME)) {
+            mockStatic(IdentityDatabaseUtil.class);
+            when(IdentityDatabaseUtil.getDBConnection(false)).thenReturn(connection1);
```

Comment:
These two lines are redundant throughout the class. How about we have a private method that takes in the connection and transaction true/false and does the job